test/functional/api/v2/hosts_controller_test.rb

```diff
@@ -301,6 +301,22 @@ def fact_json
     assert_response :success
   end
 
+  def test_rebuild_config_optimistic
+    Host.any_instance.stubs(:rebuild_config).returns({ :tftp => true, :dns => true, :dhcp => true })
+    host = FactoryGirl.create(:host)
+    post :rebuild_config, { :id => host.to_param }, set_session_user
+    assert_response :success
+    Host.any_instance.unstub(:rebuild_config)
```

Comment:
No need to unstub at the end of the method as far as I know, the method won't be stubbed outside of this test's context.